### PR TITLE
tidy up search_mirrors function

### PR DESCRIPTION
**Pull request by rnhmjoj**
Maybe I'm missing something but this seemed to be over complicated

### CODE REVIEW

File: pirate/pirate.py [hunk 1]
```diff
@@ -228,51 +228,54 @@ def combine_configs(config, args):
...
+        results = pirate.torrent.remote(
+            printer=printer,
+            pages=pages,
+            category=pirate.torrent.parse_category(printer, category),
+            sort=pirate.torrent.parse_sort(printer, sort),
+            mode=action,
+            terms=search,
+            mirror=mirror)
+    except (urllib.error.URLError, socket.timeout, IOError, ValueError):
+        printer.print('Failed', color='WARN')
```
Comment (vikstrous) [inline pirate/pirate.py]: It looks like in this case you return None? You should make that explicit if you meant to do that.

File: pirate/pirate.py [hunk 2]
```diff
@@ -228,51 +228,54 @@ def combine_configs(config, args):
...
+    else:
+        printer.print('Ok', color='alt')
+        return results, mirror
+
+
+def search_mirrors(printer, *args):
+    # try official site
+    result = connect_mirror('https://thepiratebay.mn', printer, *args)
```
Comment (vikstrous) [inline pirate/pirate.py]: change this to .org while you're at it because .mn is down

**Comment by vikstrous** [comment]
This looks like an improvement in readability. The tests seems happy, so it should be fine. Just two minor comments above.

File: pirate/pirate.py (hunk 2 above)
Comment (rnhmjoj) [inline pirate/pirate.py]: Ok

File: pirate/pirate.py (hunk 1 above)
Comment (rnhmjoj) [inline pirate/pirate.py]: Yes, that's intentional: I'll do it explicitly.

**Comment by vikstrous** [comment]
I think we should try to avoid the use of sys.exit and let the caller decide how to handle errors.

**Comment by rnhmjoj** [comment]
Raise an exception and catch that in the main, like KeyboardInterrupt?

**Comment by vikstrous** [comment]
yeah, that sounds reasonable

**Comment by rnhmjoj** [comment]
Ok?

**Comment by vikstrous** [comment]
Lgtm

**Pull request merged**